debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -128,7 +128,8 @@ public void execute(ChangeEventSourceContext context, OraclePartition partition,
             }
 
             try (LogWriterFlushStrategy flushStrategy = resolveFlushStrategy()) {
-                if (!isContinuousMining && startScn.compareTo(firstScn) < 0) {
+                if (!isContinuousMining && startScn.compareTo(firstScn.subtract(Scn.ONE)) < 0) {
+                    // startScn is the exclusive lower bound, so must be >= (firstScn - 1)
```

Comment:
Hi @chanetd, will this create any type of upgrade concern for users who already have existing offsets, or will this be seamless without any data loss?

Reply:
It should not raise any backward compatibility concerns: the meaning of
startScn hasn't changed, I just made it more explicit. And in this
particular instance, I slightly relaxed the existing sanity check because
it was too strict before due to its interpretation of startScn as an
inclusive bound.